src/org/mozilla/javascript/Parser.java

```diff
@@ -2287,14 +2294,18 @@ else if (symDeclType == Token.LP) {
         }
     }
 
-    private AstNode expr() throws IOException {
+    private AstNode expr(boolean allowTrailingComma) throws IOException {
```

Comment:
This is a minor thing but I think it's worth discussing -- rather than add a boolean parameter to this function that is used very frequently in the codebase, why not add a new method called "exprWithTrailingComma?" That would prevent having to propagate the idea of a "trailing comma" to every single place in the codebase that calls "expr", but isolate the need to understand this to the few places where we actually support that syntax.

Reply:
This is a stylistic thing but I've seen cases where over time more and more boolean flags are added to a function and it makes the logic more complicated and the function signature gets hard to understand (like you have to go back to the target function every time to understand what each boolean does).

Reply:
Right, I see that we already had a stylistic conversation about this that went the other way -- my initial thought would be to do exactly what you did and have:

> private AstNode expr() {
>   return expr(false);
> }
> 
> private AstNode expr(boolean rarelyUsedFlag) {
> // and so on

But you already went through that. Given that we already had a conversation I'm OK with this the way that it is.